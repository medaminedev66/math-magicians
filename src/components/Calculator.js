import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="Calculator">
        <span className="input">0</span>
        <span className="char">AC</span>
        <span className="char">+/-</span>
        <span className="char">%</span>
        <span className="char orange">/</span>
        <span className="char">7</span>
        <span className="char">8</span>
        <span className="char">9</span>
        <span className="char orange">*</span>
        <span className="char">4</span>
        <span className="char">5</span>
        <span className="char">6</span>
        <span className="char orange">-</span>
        <span className="char">1</span>
        <span className="char">2</span>
        <span className="char">3</span>
        <span className="char orange">+</span>
        <span className="zero">0</span>
        <span className="char">.</span>
        <span className="char orange">=</span>
      </div>
    );
  }
}

export default Calculator;
