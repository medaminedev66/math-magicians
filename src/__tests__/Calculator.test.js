import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Test Calculator component', () => {
  it('test MyCalculator function component renders without error', () => {
    const root = document.createElement('div');
    ReactDOM.render(<Calculator />, root);
  });

  it('renders correctly', () => {
    const { getByText } = render(<Calculator />);
    expect(getByText('AC')).not.toBeNull();
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('makes an operation ', () => {
    const { getByText, queryByTitle } = render(<Calculator />);
    // expect(screen.getByDisplayValue('')).toContainHTML('5');
    fireEvent.click(getByText('8'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('4'));
    fireEvent.click(getByText('='));
    expect(queryByTitle('input').textContent).toBe('12  ');
  });
});
