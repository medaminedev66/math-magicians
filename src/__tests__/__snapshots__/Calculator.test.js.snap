// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator component matches snapshot 1`] = `
<div
  className="Calculator"
>
  <span
    className="input"
    title="input"
  >
    0  
  </span>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="AC"
  >
    AC
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="+/-"
  >
    +/-
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="%"
  >
    %
  </button>
  <button
    className="char orange"
    onClick={[Function]}
    type="button"
    value="รท"
  >
    รท
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="7"
  >
    7
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="8"
  >
    8
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="9"
  >
    9
  </button>
  <button
    className="char orange"
    onClick={[Function]}
    type="button"
    value="x"
  >
    x
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="4"
  >
    4
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="5"
  >
    5
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="6"
  >
    6
  </button>
  <button
    className="char orange"
    onClick={[Function]}
    type="button"
    value="-"
  >
    -
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="1"
  >
    1
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="2"
  >
    2
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="3"
  >
    3
  </button>
  <button
    className="char orange"
    onClick={[Function]}
    type="button"
    value="+"
  >
    +
  </button>
  <button
    className="zero"
    onClick={[Function]}
    type="button"
    value="0"
  >
    0
  </button>
  <button
    className="char"
    onClick={[Function]}
    type="button"
    value="."
  >
    .
  </button>
  <button
    className="char orange"
    onClick={[Function]}
    type="button"
    value="="
  >
    =
  </button>
</div>
`;
